### DISCUSSION TITLE
Using labellers to report user content for takedown for PDS operators

### DISCUSSION THREAD

**Discussion by Bossett** [bluesky-social/atproto#2340] [General Discussion]
**Is your feature request related to a problem? Please describe.**

Related to #2226 - there should be a way for PDS owners to be alerted if material they are serving is illegal / requires urgent takedown.

**Describe the solution you'd like**

A PDS owner should be able to select a labelling service, and that service should be able to signal to the PDS that accounts/content it hosts need to be taken down / blocked from serving. The default should be the Bluesky moderation service.

This would allow the moderation process to be streamlined - so there is one kind of content takedown. Given that it affects user data, this should be a soft delete (i.e. PDS refuses to serve), but with PDS owners able to purge material.

**Comment by bnewbold** [comment]
(moved to "discussion")

We do have a plan for some additional endpoints and interactions around this. I don't have a timeline for when those will be posted/implemented, but this is mostly designed out, including:

- infrastructure-level takedown event/action in Ozone, which fans out to relevant services (PDS, AppView, Relay), using an existing background worker system with retries for reliable delivery
- ability to send mod mail (email for now) to end-users, via PDS (so mod service doesn't see/access private account email), with configurable scope (aka, delivery contingent on user config and/or PDS config, not just forwarded from any random mod service)
- ability for PDS instance to "delegate" authority to a mod service (Ozone instance). this might be a general-purpose labeler, or just a small local Ozone instance used only to administer user account (infra-level takedowns and sending mail)
- ability for PDS instances to receive admin-level "recommendations" for actions from arbitrary mod services, and have those either be forwarded as email to operators, or forwarded as a report to an Ozone instance (for easier "atproto native" centralized queuing)

IIRC an undocumented MVP of this may be in place with the current self-hosted PDS and Ozone distributions, but we are likely to tweak the endpoints and add documentation to make this more robust.
